.travis.yml

```diff
@@ -34,13 +34,13 @@ jobs:
         - ./scripts/test-setup.sh
         - export PATH=$PATH:/usr/bin/geckodriver
       script:
-        - python3 codalab_service.py test --version ${TRAVIS_BRANCH} default
-          # Make sure restarting worker doesn't cause any issues (ie in serialization/deserialization)
-        - docker restart codalab_worker_1 && python3 codalab_service.py test run --version ${TRAVIS_BRANCH}
+        - python3 test_runner.py --version ${TRAVIS_BRANCH} default
+        # Make sure restarting worker doesn't cause any issues (ie in serialization/deserialization)
+        - docker restart codalab_worker_1 && python3 test_runner.py --version ${TRAVIS_BRANCH} run
         - docker stop codalab_worker_1
-        - python3 codalab_service.py start --services worker --version ${TRAVIS_BRANCH} --shared-file-system
-        - python3 codalab_service.py test --version ${TRAVIS_BRANCH} --shared-file-system basic run read write kill resources netcat netcurl workers
-        - python3 tests/ui/ui_tester.py --headless
+        - python3 codalab_service.py start --services init worker --version ${TRAVIS_BRANCH} --shared-file-system
+        # Also run the frontend tests in this iteration of testing
+        - python3 test_runner.py --version ${TRAVIS_BRANCH} frontend basic run read write kill resources netcat netcurl workers
```

Comment:
Why the long list of tests?  Is this for `--shared-file-system`?  If so, be explicit, but I don't think you need all of these.

Reply:
I think this is to test shared file systems. This list of tests (aside from `frontend`)  was there before this change and I just copied it over. Should I maybe just run `basic`, `read`, `write`, `resources` and `workers`?

Reply:
I would have guessed that `basic` would be enough?  But maybe @bkgoksel has deeper context.

Reply:
I synced with @bkgoksel and it turns out these test modules somewhat tests shared file system code paths.

Reply:
Ok, then I guess we can leave them in for now and add a comment for why those are there.  In the future, we might want to figure out a more minimal set of tests.

Reply:
I updated the comment and created a follow-up issue for it to track the work: https://github.com/codalab/codalab-worksheets/issues/2243